modules/v1/workspaces/models/Article.php

```diff
@@ -0,0 +1,192 @@
+            [['workspace_id'], 'exist', 'skipOnError' => true, 'targetClass' => Workspace::class, 'targetAttribute' => ['workspace_id' => 'id']],
+            [['image'], 'file', 'skipOnEmpty' => true, 'extensions' => 'png, jpeg, svg, gif, jpg']
+        ];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function attributeLabels()
+    {
+        return [
+            'id' => Yii::t('app', 'ID'),
+            'parent_id' => Yii::t('app', 'Parent ID'),
+            'workspace_id' => Yii::t('app', 'Workspace ID'),
+            'title' => Yii::t('app', 'Title'),
+            'body' => Yii::t('app', 'Body'),
+            'is_folder' => Yii::t('app', 'Is Folder'),
+            'image_path' => Yii::t('app', 'Image Path'),
+            'lft' => Yii::t('app', 'Lft'),
+            'rgt' => Yii::t('app', 'Rgt'),
+            'depth' => Yii::t('app', 'Depth'),
+            'tree' => Yii::t('app', 'Tree'),
+            'created_at' => Yii::t('app', 'Created At'),
+            'updated_at' => Yii::t('app', 'Updated At'),
+            'created_by' => Yii::t('app', 'Created By'),
+            'updated_by' => Yii::t('app', 'Updated By'),
+        ];
+    }
+
+    /**
+     * @return ArticleQuery
+     */
+    public static function find()
+    {
+        return new ArticleQuery(get_called_class());
+    }
+
+    /**
+     * Gets query for [[CreatedBy]].
+     *
+     * @return ActiveQuery
+     */
+    public function getCreatedBy()
+    {
+        return $this->hasOne(User::class, ['id' => 'created_by']);
+    }
+
+    /**
+     * Gets query for [[Parent]].
+     *
+     * @return ActiveQuery
+     */
+    public function getParent()
+    {
+        return $this->hasOne(Article::class, ['id' => 'parent_id']);
+    }
+
+    /**
+     * Gets query for [[Articles]].
+     *
+     * @return ActiveQuery
+     */
+    public function getChildren()
+    {
+        return $this->hasMany(Article::class, ['parent_id' => 'id']);
+    }
+
+    /**
+     * Gets query for [[UpdatedBy]].
+     *
+     * @return ActiveQuery
+     */
+    public function getUpdatedBy()
+    {
+        return $this->hasOne(User::class, ['id' => 'updated_by']);
+    }
+
+    /**
+     * Gets query for [[Workspace]].
+     *
+     * @return ActiveQuery
+     */
+    public function getWorkspace()
+    {
+        return $this->hasOne(Workspace::class, ['id' => 'workspace_id']);
+    }
+
+    /**
+     * Delete image
+     *
+     * @return bool
+     * @throws \yii\base\ErrorException
+     */
+    public function deleteImage()
+    {
+        if ($this->image_path) {
+            $dir = dirname($this->image_path);
+            FileHelper::removeDirectory(Yii::getAlias("@storage/$dir"));
+        }
+
+        return true;
+    }
+
+    /**
+     * Check uploaded file is image
+     *
+     * @return bool
+     */
+    public function isImage()
+    {
+        return in_array($this->image->extension, ['png', 'jpeg', 'svg', 'gif', 'jpg']);
```

Comment:
@skavt As far as I remember, you use this content_type array in workspaces as well. It is better to move it to constant in general class, maybe helper and call it from there. If we will need to restrict certain image type in the system we will edit only 1 file instead of multiple.